app/utils/sentry.ts

```diff
@@ -120,128 +136,76 @@ function captureClientErrorAsBreadcrumb(error: ClientError, isFatal: boolean) {
     }
 }
 
-// Wrapper function to any calls to Sentry so that we can gather any necessary extra data
-// before sending.
-function capture(captureFunc: () => void, config?: ClientConfig) {
-    if (config?.EnableDiagnostics !== 'true') {
-        return;
-    }
-
-    try {
-        let hasUserContext = false;
-        const userContext = getUserContext();
-        if (userContext) {
-            hasUserContext = true;
-            Sentry.setUserContext(userContext);
-        }
-
-        const extraContext = getExtraContext();
-        if (Object.keys(extraContext).length) {
-            Sentry.setExtraContext(extraContext);
-        }
-
-        const buildTags = getBuildTags();
-        if (buildTags) {
-            Sentry.setTagsContext(buildTags);
-        }
-
-        if (hasUserContext) {
-            logWarning('Capturing with Sentry at ' + getDsn() + '...');
-
-            captureFunc();
-        } else {
-            logWarning('No user context, skipping capture');
-        }
-    } catch (e) {
-        // Don't want this to get into an infinite loop again...
-        logError('Exception occurred while sending to Sentry');
-        logError(e);
-    }
-}
-
-function getUserContext() {
-    // TODO: Get current user data from active database
+export const getUserContext = async (database: Database) => {
     const currentUser = {
         id: 'currentUserId',
         locale: 'en',
         roles: 'multi-server-test-role',
     };
 
-    if (!currentUser) {
-        return null;
-    }
+    const user = await getCurrentUser(database);
 
     return {
-        userID: currentUser.id, // This can be changed to id after we upgrade to Sentry >= 0.14.10,
+        userID: user?.id ?? currentUser.id,
         email: '',
         username: '',
-        extra: {
-            locale: currentUser.locale,
-            roles: currentUser.roles,
-        },
+        locale: user?.locale ?? currentUser.locale,
+        roles: user?.roles ?? currentUser.roles,
```

Comment:
do we care?

Reply:
roles and locale can come in useful.
e.g. timezone change when we got in Barbados last time.